### ISSUE TITLE
Performance issue in TreeRegionJoin

### ISSUE THREAD

**Issue by fnothaft** [bigdatagenomics/adam#1370]
Crosslinking to https://github.com/bigdatagenomics/avocado/issues/202. I'm seeing really odd performance issues in Avocado when using the Broadcast region join code. E.g., multiple order of magnitude slowdowns.

**Comment by heuermh** [comment]
@fnothaft is this still an issue?  The linked issue in Avocado was closed.

**Comment by fnothaft** [comment]
Yes, this is still an issue. We can rewrite the region join APIs to use Spark SQL, which appears to yield a large performance gain. Moving to Spark SQL would also make #1728 easier to close.
